File: src/Client/Alignment.ts
/**
 * Defines the alignment of a component.
 */
export const Alignment = Object.freeze({

	/**
	 * The element is aligned at the start.
	 */
	Start: "Start",

	/**
	 * The element is aligned in the center.
	 */
	Center: "Center",

	/**
	 * The element is aligned at the end.
	 */
	End: "End"
});

/**
 * Defines the alignment of a component.
 */
export type Alignment = typeof Alignment[keyof typeof Alignment];

File: src/Server/Alignment.cs
namespace Belin.UI.UI;

using System.Text.Json.Serialization;

/// <summary>
/// Defines the alignment of a component.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Alignment {

	/// <summary>
	/// The element is aligned at the start.
	/// </summary>
	Start,

	/// <summary>
	/// The element is aligned in the center.
	/// </summary>
	Center,

	/// <summary>
	/// The element is aligned at the end.
	/// </summary>
	End
}
